jina/peapods/pods/compoundpod.py

```diff
@@ -0,0 +1,253 @@
+        """
+        if self.is_head_router and self.replicas_args['head']:
+            return self.replicas_args['head']
+        elif not self.is_head_router and len(self.replicas_args['replicas']) == 1:
+            return self.first_pea_args
+        elif self.deducted_head:
+            return self.deducted_head
+        else:
+            raise ValueError('ambiguous head node, maybe it is deducted already?')
+
+    @head_args.setter
+    def head_args(self, args):
+        """
+        Set the arguments for the `head` of this BasePod.
+
+        :param args: arguments of the head pea
+        """
+        if self.is_head_router and self.replicas_args['head']:
+            self.replicas_args['head'] = args
+        elif not self.is_head_router and len(self.replicas_args['replicas']) == 1:
+            self.replicas_args['replicas'][0] = args  # weak reference
+        elif self.deducted_head:
+            self.deducted_head = args
+        else:
+            raise ValueError('ambiguous head node, maybe it is deducted already?')
+
+    @property
+    def tail_args(self):
+        """
+        Get the arguments for the `tail` of this BasePod.
+
+        :return: arguments of the tail pea
+        """
+        if self.is_tail_router and self.replicas_args['tail']:
+            return self.replicas_args['tail']
+        elif not self.is_tail_router and len(self.replicas_args['replicas']) == 1:
+            return self.first_pea_args
+        elif self.deducted_tail:
+            return self.deducted_tail
+        else:
+            raise ValueError('ambiguous tail node, maybe it is deducted already?')
+
+    @tail_args.setter
+    def tail_args(self, args):
+        """
+        Set the arguments for the `tail` of this BasePod.
+
+        :param args: arguments of the tail pea
+        """
+        if self.is_tail_router and self.replicas_args['tail']:
+            self.replicas_args['tail'] = args
+        elif not self.is_tail_router and len(self.replicas_args['replicas']) == 1:
+            self.replicas_args['replicas'][0] = args  # weak reference
+        elif self.deducted_tail:
+            self.deducted_tail = args
+        else:
+            raise ValueError('ambiguous tail node, maybe it is deducted already?')
+
+    @property
+    def all_args(
+        self,
+    ) -> Dict[
+        str,
+        Union[
+            List[Union[List[Namespace], Namespace, None]],
+            list,
+            List[Namespace],
+            Namespace,
+            None,
+        ],
+    ]:
+        """
+        Get all arguments of all Peas and Pods (replicas) in this CompoundPod.
+
+        :return: arguments for all Peas and pods
+        """
+        args = {
+            'peas': ([self.replicas_args['head']] if self.replicas_args['head'] else [])
+            + ([self.replicas_args['tail']] if self.replicas_args['tail'] else []),
+            'replicas': self.replicas_args['replicas'],
+        }
+        return args
+
+    @property
```

Comment:
Is this property needed for this Pod class?

Reply:
and if it is needed, shouldnt it be the sum of all the inner Pods num peas?

Reply:
The `num_peas` function of the `BaseFlow` needs this function.

Reply:
and what is it used for?